src/vdom/component-recycler.js

```diff
@@ -33,13 +33,13 @@ export function createComponent(Ctor, props, context) {
 		inst;
 
 	if (Ctor.prototype && Ctor.prototype.render) {
-		inst = new Ctor(props, context);
+		inst = new /** @type {(props, context) => void} */(Ctor)(props, context);
 		Component.call(inst, props, context);
 	}
 	else {
 		inst = new Component(props, context);
 		inst.constructor = Ctor;
-		inst.render = doRender;
+		inst['render'] = doRender;
```

Comment:
Seems like an unnecessary change here

Reply:
The problem here is that `render` is being dynamically tagged on to the `inst` object. Since it doesn't exist on the object, any type checker will give the following error:

<img width="692" alt="render" src="https://user-images.githubusercontent.com/249218/42736418-74f5916c-881b-11e8-99b8-76a0894e5ca5.png">

You can get around this in one of two ways: type casting the `inst` object to something more generic, such as `any` (worst choice). Or simply escaping the expando property. During minification the `['']` gets removed and the property is `inst.render`. So it doesn't affect the final output. The brackets and quotes basically hide the expando property from the type checker.

This is a very common problem with dealing with expando properties and type systems. TypeScript users prefer to escape them as I have done.